lib/ansible/modules/network/bigswitch/bcf_switch.py

```diff
@@ -0,0 +1,185 @@
+            "validate_certs": false
+        }
+    }
+}
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.bigswitch_utils import Rest, Response
+from ansible.module_utils.pycompat24 import get_exception
+
+
+def leaf_switch(module):
+    try:
+        access_token = module.params['access_token'] or os.environ['BIGSWITCH_ACCESS_TOKEN']
+    except KeyError:
+        e = get_exception()
+        module.fail_json(msg='Unable to load %s' % e.message)
+
+    name = module.params['name']
+    fabric_role = module.params['fabric_role']
+    leaf_group = module.params['leaf_group']
+    dpid = '00:00:' + module.params['mac']
+    state = module.params['state']
+    controller = module.params['controller']
+
+    rest = Rest(module,
+                {'content-type': 'application/json', 'Cookie': 'session_cookie=' + access_token},
+                'https://' + controller + ':8443/api/v1/data/controller/core')
+
+    response = rest.get('switch-config', data={})
+    if response.status_code != 200:
+        module.fail_json(msg="failed to obtain existing switch config: {}".format(response.json['description']))
+
+    config_present = False
+    for switch in response.json:
+        if all((switch['name'] == name,
+                switch['fabric-role'] == fabric_role,
+                switch['dpid'] == dpid)):
+            config_present = switch.get('leaf-group', None) == leaf_group
+            if config_present:
+                break
+
+    if state in ('present') and config_present:
+        module.exit_json(changed=False)
+
+    if state in ('absent') and not config_present:
+        module.exit_json(changed=False)
+
+    if state in ('present'):
+        data = {'name': name, 'fabric-role': fabric_role, 'leaf-group': leaf_group, 'dpid': dpid}
+        response = rest.put('switch-config[name="%s"]' % name, data)
+        if response.status_code == 204:
+            module.exit_json(changed=True)
+        else:
+            module.fail_json(msg="error configuring switch '{}': {}".format(name, response.json['description']))
+
+    if state in ('absent'):
+        response = rest.delete('switch-config[name="%s"]' % name, data={})
+        if response.status_code == 204:
+            module.exit_json(changed=True)
+        else:
+            module.fail_json(msg="error deleting switch '{}': {}".format(name, response.json['description']))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type='str', required=True),
+            fabric_role=dict(choices=['spine', 'leaf'], required=True),
+            leaf_group=dict(type='str', required=False),
+            mac=dict(type='str', required=True),
+            controller=dict(type='str', required=True),
+            state=dict(choices=['present', 'absent'], default='present'),
+            validate_certs=dict(type='bool', default='True'),
+            access_token=dict(type='str', no_log=True)
+        )
+    )
```

Comment:
Can be implemented now as the module is already idempotent.

But before, what does this mean?

lib/ansible/modules/network/bigswitch/bcf_switch.py:0:0: E319 RETURN.changed.: expected a dictionary
lib/ansible/modules/network/bigswitch/bcf_switch.py:0:0: E319 RETURN.invocation.description: required key not provided @ data['description']. Got None
lib/ansible/modules/network/bigswitch/bcf_switch.py:0:0: E319 RETURN.invocation.module_args: extra keys not allowed @ data['module_args']. Got {'validate_certs': False, 'fabric_role': 'leaf', 'controller': '192.168.86.230', 'mac': '00:00:00:02:00:02', 'name': 'Rack1Leaf1', 'state': 'present', 'access_token': None, 'leaf_group': 'R1'}
lib/ansible/modules/network/bigswitch/bcf_switch.py:0:0: E319 RETURN.invocation.returned: required key not provided @ data['returned']. Got None
lib/ansible/modules/network/bigswitch/bcf_switch.py:0:0: E319 RETURN.invocation.type: required key not provided @ data['type']. Got None
click here for bot help